vtex-sites/nextjs.store
# feat: Incremental Static Build

## What's the purpose of this pull request?
This PR uses the new WebOps capability called Incremental Static Builds. This should drastically improve response times and reduce infrastructure usage on deferred pages, like PDPs and PLPs

## How does it work?
WebOps recently released this Incremental Static Build feature. To use on Next.JS, make sure to use `getStaticProps` and `getStaticPaths` APIS and enable `USE_STALE_CACHE=true` on `vtex.env`.

## How to test it?
Open a PDP. Check the response header called `x-faststore-cache`. The first request should give you a `MISS`. The second request should give you `HIT`. Note that `HIT`s are much faster than `MISS`. Also, note that all subsequent requests have a HIT as value. 

## References
- Original PR on WebOps: https://github.com/vtex/vtex-cicd-platform/pull/105
